osu.Game/Online/RealtimeMultiplayer/MultiplayerRoomSettings.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using JetBrains.Annotations;
+using osu.Game.Online.API;
+
+namespace osu.Game.Online.RealtimeMultiplayer
+{
+    [Serializable]
+    public class MultiplayerRoomSettings : IEquatable<MultiplayerRoomSettings>
+    {
+        public int? BeatmapID { get; set; }
+
+        public int? RulesetID { get; set; }
+
+        [NotNull]
+        public IEnumerable<APIMod> Mods { get; set; } = Enumerable.Empty<APIMod>();
+
+        public bool Equals(MultiplayerRoomSettings other) => BeatmapID == other.BeatmapID && Mods.SequenceEqual(other.Mods) && RulesetID == other.RulesetID;
```

Comment:
Going to have to be careful about order of `Mods` with this comparer. Maybe the setter of `Mods` should ensure sorting?

Reply:
Hmm... I think it's safe enough for now as worst-case it's just going to trigger an update of settings to the same settings. I copied this from the existing implementation in `SpectatorState` for what that's worth.